ext/openssl/openssl.c

```diff
@@ -881,7 +890,7 @@ static void add_assoc_name_entry(zval * val, char * key, X509_NAME * name, int s
 
 		if (needs_free) {
 			/* ASN1_STRING_to_UTF8(3): The buffer out should be freed using free(3) */
-			OPENSSL_free(to_add);
+			OPENSSL_free((char *)to_add);
```

Comment:
It might be better to initialize `to_add_buf` in the same block as `to_add` and then we could just test if `to_add_buf != NULL` and free it then. You can then remove `needs free`...